Builder/ElasticaDatagridBuilder.php

```diff
@@ -106,9 +106,15 @@ public function getBaseDatagrid(AdminInterface $admin, array $values = array())
             $defaultOptions
         );
 
-        $proxyQuery = new ElasticaProxyQuery(
-            $this->finderProvider->getFinderByAdmin($admin)
-        );
+        $proxyQuery = $admin->createQuery ();
+        // if the default modelmanager query builder is used, we need to replace it with elastica
+        // if not, that means $admin->createQuery has been overriden by the user and return already an ElasticaProxyQuery object
+        if (! $query instanceof Sonata\AdminSearchBundle\ProxyQuery\ElasticaProxyQuery) {
+            $issmart = $this->isSmart ( $admin, $values );
+            if ($issmart) {
```

Comment:
Please do not create variables that are used only once. If you need to clarify what something does, please use comments. In that case, the name of the variable and the method is the same so you can simply do : `if ($this->isSmart($admin, $values))`.